### PR TITLE
osdmaptool/cleanup: Completed osdmaptool's usage

### PR DESCRIPTION
**Pull request by nvedant07**
Fixes: http://tracker.ceph.com/issues/3214
Signed-off-by: Vedant Nanda <vedant15114@iiitd.ac.in>

### CODE REVIEW

File: src/tools/osdmaptool.cc [hunk 1]
```diff
@@ -25,18 +25,25 @@ using namespace std;
...
+  cout << "   --test-map-pgs-dump [--pool <poolid>] [--range-first <first> --range-last <last>] map all pgs" << std::endl;
+  cout << "   --test-map-pgs-dump-all [--pool <poolid>] [--range-first <first> --range-last <last>] map all pgs to osds" << std::endl;
   cout << "   --mark-up-in            mark osds up and in (but do not persist)" << std::endl;
   cout << "   --clear-temp            clear pg_temp and primary_temp" << std::endl;
   cout << "   --test-random           do random placements" << std::endl;
   cout << "   --test-map-pg <pgid>    map a pgid to osds" << std::endl;
   cout << "   --test-map-object <objectname> [--pool <poolid>] map an object to osds"
        << std::endl;
+  cout << "   --clobber               allows osdmaptool to overwrite <mapfilename> if it already exists" << std::endl;
+  cout << "   --dump <type>           displays result in plain text when <type> is 'plain', json for any other argument." << std::endl;
```
Comment (tchaikov) [inline src/tools/osdmaptool.cc]: would be better to use "format" as the name of the metavar 

i don't really understand this. maybe we can put
> displays result in plain text when <type> is 'plain', 'json' if the specified type is not supported.

File: src/tools/osdmaptool.cc [hunk 2]
```diff
@@ -25,18 +25,25 @@ using namespace std;
...
+  cout << "   --tree                  displays a tree of the map" << std::endl;
+  cout << "   --pgp-bits <bits>       pgp_num map attribute will be shifted by <bits>" << std::endl;
+  cout << "   --create-from-conf      creates an osd map with default configurations" << std::endl;
```
Comment (tchaikov) [inline src/tools/osdmaptool.cc]: might want to move "--createsimple <numosd> [--clobber] [--pg_bits <bitsperosd>]]" down to next to "--create-from-conf" as osdmaptool does not "createsimple" by default.

Comment (tchaikov) [inline src/tools/osdmaptool.cc]: and could you please update the `doc/man/8/osdmaptool.rst` accordingly?

**Comment by tchaikov** [comment]
and you might want to fold your commits in this PR into a single one.

File: src/tools/osdmaptool.cc (hunk 1 above)
Comment (nvedant07) [inline src/tools/osdmaptool.cc]: Sure ill change it to format.

`osdmaptool --dump plain osdmap` prints plain text while `osdmaptool --dump plai osdmap` prints json. Your line makes that more clear. I'll update that too. Thanks!

File: src/tools/osdmaptool.cc (hunk 2 above)
Comment (nvedant07) [inline src/tools/osdmaptool.cc]: Sure i'll do that.

I'll move the `--pg-bits` next to `--pgp-bits` as well then.

**Comment by nvedant07** [comment]
@tchaikov  done

File: doc/man/8/osdmaptool.rst [hunk 3]
```diff
@@ -29,6 +29,11 @@ Options
    will simply make the tool print a plaintext dump of the map, after
    any modifications are made.
 
+.. option:: --dump <format>
+
+   displays the map in plain text when <format> is 'plain', 'json' if specified
```
Comment (amitkumar50) [inline doc/man/8/osdmaptool.rst]: @nvedant07 NIT If this is a new sentence. How good to start with Capital letter?
My suggestion..
Your thoughts?

File: doc/man/8/osdmaptool.rst [hunk 4]
```diff
@@ -42,22 +47,75 @@ Options
...
-.. option:: --test-map-pgs [--pool poolid]
+   creates an osd map with default configurations.
+
+.. option:: --test-map-pgs [--pool poolid] [--range-first <first> --range-last <last>]
 
    will print out the mappings from placement groups to OSDs.
+   If range is specified, then it iterates from first to last in the directory 
+   specified by argument to osdmaptool.
+   Eg: **osdmaptool --test-map-pgs --range-first 0 --range-last 2 osdmap_dir**.
+   This will iterate thorugh the files named 0,1,2 in osdmap_dir.
```
Comment (amitkumar50) [inline doc/man/8/osdmaptool.rst]: @nvedant07 Wrong spelling 'thorugh'

File: doc/man/8/osdmaptool.rst [hunk 5]
```diff
@@ -42,22 +47,75 @@ Options
...
+
+.. option:: --test-map-pgs-dump-all [--pool poolid] [--range-first <first> --range-last <last>]
 
    will print out the summary of all placement groups and the mappings
-   from them to the mapped OSDs.
+   from them to all the OSDs.
+   If range is specified, then it iterates from first to last in the directory
+   specified by argument to osdmaptool.
+   Eg: **osdmaptool --test-map-pgs-dump-all --range-first 0 --range-last 2 osdmap_dir**.
+   This will iterate thorugh the files named 0,1,2 in osdmap_dir.
```
Comment (amitkumar50) [inline doc/man/8/osdmaptool.rst]: @nvedant07 Again 'thorugh'

File: doc/man/8/osdmaptool.rst [hunk 6]
```diff
@@ -42,22 +47,75 @@ Options
...
+
+   map a particular placement group(specified by objectname) to the OSDs.
+
+.. option:: --test-crush [--range-first <first> --range-last <last>]
+
+   map placement groups to acting OSDs.
+   If range is specified, then it iterates from first to last in the directory
+   specified by argument to osdmaptool.
+   Eg: **osdmaptool --test-crush --range-first 0 --range-last 2 osdmap_dir**.
+   This will iterate thorugh the files named 0,1,2 in osdmap_dir.
```
Comment (amitkumar50) [inline doc/man/8/osdmaptool.rst]: @nvedant07 same

File: doc/man/8/osdmaptool.rst (hunk 3 above)
Comment (smithfarm) [reply doc/man/8/osdmaptool.rst]: @amitkumar50 I don't agree with this comment. The format used is consistent with the rest of the document.

**Comment by smithfarm** [comment]
@nvedant07 Can you please rebase this PR and fix the spelling of the word "through" (as in "iterate through")?

**Comment by nvedant07** [comment]
Thank you for pointing out @amitkumar50 @smithfarm. I have fixed the spelling and rebased this PR. Please have a look

**Comment by gregsfortytwo** [comment]
I didn't look too closely but I presume this is good based on previous reviews and comments. Let's see if it works now that I resolved a merge conflict.

**Comment by tchaikov** [comment]
@gregsfortytwo in future, we might want to avoid unnecessary merge commits like 3e223763532ca2a8aa327a9c4f2703e83f5e91ff .

**Pull request merged**